extensions/aris/src/bigDataCluster/tree/controllerTreeNode.ts

```diff
@@ -0,0 +1,475 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+'use strict';
+
+import * as vscode from 'vscode';
+import * as azdata from 'azdata';
+import * as nls from 'vscode-nls';
+import { IControllerTreeChangeHandler } from './controllerTreeChangeHandler';
+import { TreeNode } from './treeNode';
+import { IEndPoint, IControllerError } from '../controller/types';
+import { ArisController } from '../controller/controller';
+import { IconPath, BdcItemType } from '../constants';
+
+const localize = nls.loadMessageBundle();
+
+export abstract class ControllerTreeNode extends TreeNode {
+	private _description: string;
+	private _nodeType: string;
+	private _iconPath: { dark: string, light: string };
+	private _treeChangeHandler: IControllerTreeChangeHandler;
+
+	constructor(p?: {
```

Comment:
maybe defining a local class makes sense rather than repeat definitions of this collection of values makes sense.  In general I'd recommend using classes rather than repeat definitions.

Reply:
no repeat any more.